### PR TITLE
chore(ext): Extensions wire

### PR DESCRIPTION
**Pull request by InfamousVague**
Need help getting this over the line; remaining tasks are: 

- [x] Update the chatbar properly when the draft changes for that chat.
- [x] Investigate events not firing correctly when clicking emoji on the emoji picker extension. 
- [x] Wire the navigation at the bottom to properly scroll to the correct emoji section. (this may require JS for now, not sure.)
- [x] Resolve state not saving the enabled / disabled state of extensions.

### CODE REVIEW

**Comment by InfamousVague** [comment]
>  Wire the navigation at the bottom to properly scroll to the correct emoji section. (this may require JS for now, not sure.) 

This step is not done but I'd like to move on to other things for now that are more important to the pending release date.

**Comment by phillsatellite** [comment]
@InfamousVague Would you like all these to be tickets? 

1. if you backspace all of your types emojies than select a new one, the old ones will re appear when I select the new emoji 

https://user-images.githubusercontent.com/93608357/225110074-b404d32e-b986-4e6e-84e1-d1c416af562f.mov

2. If you are in edit message modal and you are adding emojis, it will add them to the regular input box for sending messages 

https://user-images.githubusercontent.com/93608357/225110571-d2c9ffbf-998c-42a1-a252-901bc6b714da.mov

3. When you add emoji's the first one will be cutoff until you add more

https://user-images.githubusercontent.com/93608357/225110666-90e52283-67b2-41fc-83e7-a662eeb2e4c3.mov

3. Emoji will only appear in chat if you have text before it

https://user-images.githubusercontent.com/93608357/225110905-5adeab63-0ffb-47db-87fd-36d98f9b8c51.mov

4. We should prob have the emoji selector close when the user clicks back into the input box 

https://user-images.githubusercontent.com/93608357/225111096-36bf3ef7-a1aa-4045-8878-8150b67bc1ef.mov

**Comment by InfamousVague** [comment]
@dariusc93 could the emoji being removed be something on the warp side ?

**Comment by dariusc93** [comment]
> @dariusc93 could the emoji being removed be something on the warp side ?

It wouldnt be. It doesnt extract or remove anything in message as the only checks in place are for the length (which been in place even during the old uplink)
image

I can try to reproduce the issue @phillsatellite have in uplink in this pr and get the output of the message and see if the unicode is included. It may not be rendering it out for some reason

**Comment by dariusc93** [comment]
I was able to reproduce it in uplink and debugging it, it looks like its not apart of the message before it is even passed into raygun to send, so it may be something further up either stripping it out, or its only being rendered but not included in the message itself. 

```
2023-03-14 22:12:01 | DEBUG | Message Sent: ["hello world"]
2023-03-14 22:12:01 | DEBUG | received message event: MessageSent { conversation_id: 24e8f06b-f908-77ce-851a-f7d8b9e7900f, message_id: a675c3b5-57a9-48c7-a00f-79e801ec9dc4 }
2023-03-14 22:12:01 | DEBUG | got event: MessageSent { conversation_id: 24e8f06b-f908-77ce-851a-f7d8b9e7900f, message_id: a675c3b5-57a9-48c7-a00f-79e801ec9dc4 }
2023-03-14 22:12:01 | DEBUG | Message { id: a675c3b5-57a9-48c7-a00f-79e801ec9dc4, message_type: Message, conversation_id: 24e8f06b-f908-77ce-851a-f7d8b9e7900f, sender: DID("z6MktPHDTxD4rJxX68cmdar22FJ7CrKxHPYmnxzLe4LchTy1"), date: 2023-03-15T02:12:01.672625694Z, modified: None, pinned: false, reactions: [], replied: None, value: ["hello world"], attachment: [], signature: Some([106, 8, 85, 170, 248, 156, 170, 78, 68, 185, 55, 208, 62, 152, 161, 57, 0, 208, 0, 191, 236, 121, 135, 126, 95, 100, 219, 34, 252, 119, 100, 176, 192, 47, 172, 193, 97, 80, 141, 19, 132, 126, 163, 24, 187, 27, 20, 27, 205, 176, 244, 202, 59, 64, 54, 108, 202, 43, 25, 244, 191, 55, 139, 12]), metadata: {} }
```

**Comment by lgmarchi** [comment]
> @InfamousVague Would you like all these to be tickets?
> 
> 1. if you backspace all of your types emojies than select a new one, the old ones will re appear when I select the new emoji
> 
> 2. If you are in edit message modal and you are adding emojis, it will add them to the regular input box for sending messages
> 
> 3. When you add emoji's the first one will be cutoff until you add more
> 
> 3. Emoji will only appear in chat if you have text before it
> 
> 4. We should prob have the emoji selector close when the user clicks back into the input box

@phillsatellite, just second point, will be necessary to use Emoji Picker in message, as it works on Discord, maybe a ticket just for this, would be nice.

All other points were fixed!

File: kit/src/elements/input/mod.rs [hunk 1]
```diff
@@ -131,7 +131,10 @@ pub struct Props<'a> {
     aria_label: Option<String>,
     is_password: Option<bool>,
     disabled: Option<bool>,
+    #[props(optional)]
```
Comment (sdwoodbury) [inline kit/src/elements/input/mod.rs:134]: why make an effort to add this back? it doesn't do anything

Comment (InfamousVague) [reply kit/src/elements/input/mod.rs:134]: Didn't know it does nothing until I read your comment on another PR the other day 😂

**Review by sdwoodbury — COMMENTED** [review]
I noticed that the emoji selector closes after selecting an emoji. If one wants to select multiple emojis, this is undesirable.

**Comment by lgmarchi** [comment]
> I noticed that the emoji selector closes after selecting an emoji. If one wants to select multiple emojis, this is undesirable.

This behavior to close is like Discord, but we can change it without problems

**Comment by dariusc93** [comment]
> I noticed that the emoji selector closes after selecting an emoji. If one wants to select multiple emojis, this is undesirable.

~~[This should have fixed it](https://github.com/Satellite-im/Uplink/pull/409/commits/b32ab36cb16f85afb5b51212a5fe86af991e51de) as what was discussed on discord unless more changes been done to it since then~~

Err ignore that. Yes it should probably have something to keep it open for multiple items if that is desirable

**Comment by InfamousVague** [comment]
> > I noticed that the emoji selector closes after selecting an emoji. If one wants to select multiple emojis, this is undesirable.
> 
> ~[This should have fixed it](https://github.com/Satellite-im/Uplink/pull/409/commits/b32ab36cb16f85afb5b51212a5fe86af991e51de) as what was discussed on discord unless more changes been done to it since then~
> 
> Err ignore that. Yes it should probably have something to keep it open for multiple items if that is desirable

I think we can leave it like that for now, it's intentional and how most other apps function.

**Comment by phillsatellite** [comment]
@lgmarchi @sdwoodbury Lucas wanted me to share what I found as well while testing, I spent a good amount of time on here I was able to replicate both of the issues that Lucas had mentioned in the Dev chat I also found 2 others as well (I think the first one is the same as Lucas found ("2. If we backspace first char and try to put new emoji it will back char that we removed"). I figured these will prob be made into tickets but I still wanted to leave my findings 

When you type some text in the input box and then open emoji selector it will clear out the text you had typed (until you select the emoji you want, if you click out of the emoji selector instead of selecting an emoji the text that was in the input box will remain missing)

https://user-images.githubusercontent.com/93608357/225776879-c3a8c386-e505-456e-ac46-0382c2f8daa8.mov

Seems like this emoji is cutoff as well 

Screenshot 2023-03-16 at 7 24 56 PM

**Comment by sdwoodbury** [comment]
> @lgmarchi @sdwoodbury Lucas wanted me to share what I found as well while testing, I spent a good amount of time on here I was able to replicate both of the issues that Lucas had mentioned in the Dev chat I also found 2 others as well (I think the first one is the same as Lucas found ("2. If we backspace first char and try to put new emoji it will back char that we removed"). I figured these will prob be made into tickets but I still wanted to leave my findings
> 
> When you type some text in the input box and then open emoji selector it will clear out the text you had typed (until you select the emoji you want, if you click out of the emoji selector instead of selecting an emoji the text that was in the input box will remain missing)
> 
> Seems like this emoji is cutoff as well
> 
> Screenshot 2023-03-16 at 7 24 56 PM

I believe I fixed this with commit [c6db702](https://github.com/Satellite-im/Uplink/pull/409/commits/c6db7022f5f29faa2fa5d27c7f463c61170f60ec)

**Comment by InfamousVague** [comment]
the last emoji being cut off is just how that emoji looks 🫡

**Comment by lgmarchi** [comment]
Navigation on emoji selector is fixed. 


https://user-images.githubusercontent.com/63157656/225978852-3c76d05b-88c2-4824-8c12-c3c7f27a6697.mov

**Review by phillsatellite — DISMISSED** [review]

**Review by sdwoodbury — DISMISSED** [review]

**Review by sdwoodbury — APPROVED** [review]

**Pull request merged**